### ISSUE TITLE
Allowing generator for library

### ISSUE THREAD

**Issue by a14n** [kevmoo/source_gen.dart#38]
It seems that `LibraryElement` are not exposed to `Generator`.

```dart
class LibGenerator extends Generator {
  const LibGenerator();

  Future<String> generate(Element element) async {
    if (element is! LibraryElement) return null;
    // never here because LibraryElement is not exposed
    return "// I'm generated for lib";
  }
}
```

**Comment by kevmoo** [comment]
Add a method to `Generator` that takes in the library element and defaults to returning `null`.

**Comment by kevmoo** [comment]
Now published with v0.2.2

Thanks again!

**Comment by a14n** [comment]
Cool !
